File: src/test/java/org/simpleframework/xml/transform/sql/TimestampTransformTest.java
package org.simpleframework.xml.transform.sql;

import java.sql.Timestamp;
import junit.framework.TestCase;

public class TimestampTransformTest extends TestCase {
   
   public void testTimestamp() throws Exception {
      long now = System.currentTimeMillis();
      Timestamp date = new Timestamp(now);
      TimestampTransform format = new TimestampTransform();
      String value = format.write(date);
      Timestamp copy = format.read(value);
      
      assertEquals(date, copy);  
      assertEquals(copy.getTime(), now);
   }
}

File: trunk/download/stream/port/src/main/Xml/Transform/TimestampTransformTest.cs
#region Using directives
using SimpleFramework.Xml.Transform;
using SimpleFramework.Xml;
using System.Collections.Generic;
using System;
#endregion
package SimpleFramework.Xml.transform;
public class TimestampTransformTest : TestCase {
   public void TestTimestamp() {
      long now = System.currentTimeMillis();
      Timestamp date = new Timestamp(now);
      DateTransform format = new DateTransform(Timestamp.class);
      String value = format.write(date);
      Date copy = format.read(value);
      AssertEquals(date, copy);
      AssertEquals(copy.getTime(), now);
      assertTrue(copy instanceof Timestamp);
   }
}
}
